api/rest/client/client.go

```diff
@@ -92,10 +162,10 @@ type Client struct {
 	p2p       host.Host
 }
 
-// NewClient initializes a client given a Config.
-func NewClient(cfg *Config) (*Client, error) {
+// NewDefaultClient initializes a client given a Config.
+func NewDefaultClient(cfg *Config) (*DefaultClient, error) {
```

Comment:
and here you should be able to return `Client` (if `defaultClient` correctly implements the interface)